docs/viplab3.0/computation_result.md

```diff
@@ -166,28 +167,29 @@ In Viplab 2.0, the following MIMEtypes where allowed. They contain some custom d
 
 
 #### notifications artifact JSON object
 Each notifications artifact contains at least one summary and unlimited number of notification elemets providing more details
 
-|Key            |Value Type |Opt / Must |Description |Comment | AS |
-|---------------|-----------|-----------|------------|--------|----|
+|Key            |Value Type |Opt / Must |Description |Comment |
+|---------------|-----------|-----------|------------|--------|
 | summary | string | must | summary of all info_element's: e.g. one warning, one error may have a summary "An error occured." | E.g.: "[Error] Backend has detected an error: no result!" or "Success!" |
-| elements |  array of notification objects | opt | the more detailed notification objects for this notifications object| |
+| notifications |  array of notification objects | opt | the more detailed notification objects for this notifications object| |
 
 
-#### notification artifact JSON object
-|Key            |Value Type |Opt / Must |Description |Comment | AS |
-|---------------|-----------|-----------|------------|--------|----|
+#### notification JSON object
+|Key            |Value Type |Opt / Must |Description |Comment |
+|---------------|-----------|-----------|------------|--------|
 | severity | string | must | One of {"error", "warning", "info"} | | |
-| type | string | must | One of {"system", "chain", "output", "callcheck", "interpreter", "compiler", "linker", "executable"} | For all chains: {"system, "chain", "output"}; for Octave/Matlab: {"callcheck", "interpreter"}; for C: {"callcheck", "compiler", "linker", "executable"}; for DuMuX: {"executable"} | If type equals "system", *whole* Result if of interest for a bug report. | Java?? |
+| type | string | must | One of {"system", "chain", "output", "callcheck", "interpreter", "compiler", "linker", "executable"} | For all chains: {"system, "chain", "output"}; for Octave/Matlab: {"callcheck", "interpreter"}; for C: {"callcheck", "compiler", "linker", "executable"}; for Java: {"callcheck", "compiler", "executable"}; for DuMuX: {"executable"}  If type equals "system", *whole* Result if of interest for a bug report. |
 | message | string | must | summary of one message to the user; it should not contain wrong error locations (file, line, col). Together with "source"/"line" or "col" (containing corrected locations) it should give the most interesting info. | may be empty string (',' in optional location_part) |
-| origin  | origin json object | opt | If the message can be linked to part from the ComputationTemplate, the original position can be found inside | only allowed if "type" in {"compiler", "interpreter", "callcheck"} | 
-| output | output | opt | the original message from which this notification has be extracted from | | 
+| origin  | origin json object | opt | If the message can be linked to part from the ComputationTask, the original position can be found inside | only allowed if "type" in {"compiler", "interpreter", "callcheck"} |
```

Comment:
What if it can't compile the ComputationTemplate parts?

Reply:
We need to have both output and origin. The callcheck will e.g. only create a notification with an origin when it parses the code and finds illegal calls.
The case you describe might have both, origin and output. The notification will be caused by parsing the output and extracting the message and origin will then link the positions in the output to the correct location in the part, which might be different if there are e.g. hidden code parts.
Output only might also appear if e.g. there is a program that detect something stdout or stderr which does not have any relation to any of the parts. In the container case, this might that the container cannot be found.

Reply:
ah ok